src/main/java/org/veupathdb/service/eda/ds/plugin/pass/BoxplotPlugin.java

```diff
@@ -101,18 +103,31 @@ protected void writeResults(OutputStream out, Map<String, InputStream> dataStrea
     String showMissingness = spec.getShowMissingness() != null ? spec.getShowMissingness().getValue() : "FALSE";
     String computeStats = spec.getComputeStats() != null ? spec.getComputeStats().getValue() : "FALSE";
     String showMean = spec.getMean() != null ? spec.getMean().getValue() : "FALSE";
+    String showPoints = spec.getPoints().getValue();
     
-    useRConnectionWithRemoteFiles(dataStreams, connection -> {
-      connection.voidEval(getVoidEvalFreadCommand(DEFAULT_SINGLE_STREAM_NAME, 
+    List<String> nonStrataVarColNames = new ArrayList<String>();
+    nonStrataVarColNames.add(toColNameOrEmpty(spec.getXAxisVariable()));
+    nonStrataVarColNames.add(toColNameOrEmpty(spec.getYAxisVariable()));
+
+    RFileSetProcessor filesProcessor = new RFileSetProcessor(dataStreams)
+      .add(DEFAULT_SINGLE_STREAM_NAME, 
+        spec.getMaxAllowedDataPoints(), 
+        showMissingness, 
+        nonStrataVarColNames, 
+        (name, conn) ->
+        conn.voidEval(getVoidEvalFreadCommand(DEFAULT_SINGLE_STREAM_NAME, 
```

Comment:
A little cleaner here to use the parameter 'name' than the constant again; works, but makes it less clear that we could potentially use this bifunction elsewhere (with other file names).